libsolidity/ast/ASTAnnotations.h

```diff
@@ -108,6 +108,7 @@ struct VariableDeclarationAnnotation: ASTAnnotation
 
 struct StatementAnnotation: ASTAnnotation, DocumentedAnnotation
 {
+	bool isInLoop = false;
```

Comment:
Do we really need this as part of the annotations? As it is a kind of transient information (only relevant during syntax checking), I think it would be better to put it as a member of `SyntaxChecker`.